server/api.go

```diff
@@ -45,12 +50,58 @@ type PRDetails struct {
 	Reviews            []*github.PullRequestReview `json:"reviews"`
 }
 
+type HTTPHandlerFuncWithUser func(w http.ResponseWriter, r *http.Request, userID string) http.HandlerFunc
+
 func writeAPIError(w http.ResponseWriter, err *APIErrorResponse) {
 	b, _ := json.Marshal(err)
 	w.WriteHeader(err.StatusCode)
 	w.Write(b)
 }
 
+func (p *Plugin) initialiseAPI() {
+	p.router = mux.NewRouter()
+
+	webHookRouter := p.router.PathPrefix(API_WEBHOOK).Subrouter()
+	oauthRouter := p.router.PathPrefix(API_OAUTH).Subrouter()
+	apiRouter := p.router.PathPrefix(API_PREFIX).Subrouter()
+
+	webHookRouter.HandleFunc("/", p.handleWebhook)
+
+	oauthRouter.HandleFunc("/connect", p.extractUserMiddleWare(p.connectUserToGitHub, false))
+	oauthRouter.HandleFunc("/complete", p.extractUserMiddleWare(p.completeConnectUserToGitHub, false))
+
+	apiRouter.HandleFunc("/connected", p.extractUserMiddleWare(p.getConnected, true)).Methods("GET")
+	apiRouter.HandleFunc("/todo", p.extractUserMiddleWare(p.postToDo, true))
```

Comment:
Why does this method lack a `http method`?

Reply:
I missed your comment at https://github.com/mattermost/mattermost-plugin-github/pull/248#issuecomment-615868851.

- `/webhook` uses `POST`
- `/oauth/connect` uses `GET`
- `/oauth/complete` uses `GET`
- `/api/v1/todo` uses `POST`

Reply:
@hanzei sorry I forgot `/api/v1/settings` is it `GET`?

Reply:
Yes